parley/src/inline_box.rs

```diff
@@ -0,0 +1,16 @@
+// Copyright 2024 the Parley Authors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+/// A box to be laid out inline with text
+#[derive(Debug, Clone)]
+pub struct InlineBox {
+    /// User-specified identifier for the box, which can be used by the user to determine which box in
+    /// parley's output corresponds to which box in it's input.
+    pub id: u64,
+    /// The character index into the underlying text string at which the box should be placed.
```

Comment:
We should document the bidi decision here (i.e. that the boxes follow the bidi of the logically previous character)